src/mraid.ts

```diff
@@ -58,4 +63,24 @@ export class MRAIDImplementation implements MRAID {
   open(url: URL) {
     console.log(`open(), url -> ${url}`);
   }
+
+  // #endregion
+
+  // #region SDKApi
+
+  notifyReady() {
+    this.updateState(MraidState.Default);
+    this.eventsCoordinator.fireReadyEvent();
+  }
+
+  notifyError(message: string, action: string | null): void {
+    this.eventsCoordinator.fireErrorEvent(message, action);
+  }
+
+  // #endregion
+
+  private updateState(newState: MraidState) {
+    this.currentState = newState;
```

Comment:
Don't you need to validate that the change from `currentState` to `newState` is valid ?

Reply:
Currently we have only one state transition implemented (loading -> default). So i added check in `notifyReady` method. Will rework code if needed when we add other transitions